zebra-state/src/service.rs

```diff
@@ -29,14 +34,104 @@ struct StateService {
     /// Holds data relating to finalized chain state.
     sled: FinalizedState,
     /// Holds data relating to non-finalized chain state.
-    _mem: NonFinalizedState,
+    mem: NonFinalizedState,
+    /// Blocks awaiting their parent blocks for contextual verification.
+    queued_blocks: QueuedBlocks,
+}
+
+#[derive(Debug, Error)]
+#[error("block is not contextually valid")]
+struct CommitError(#[from] ValidateContextError);
+
+#[derive(displaydoc::Display, Debug, Error)]
+enum ValidateContextError {
+    /// block.height is lower than the current finalized height
+    OrphanedBlock,
 }
 
 impl StateService {
+    const REORG_LIMIT: block::Height = crate::constants::MAX_BLOCK_REORG_HEIGHT;
```

Comment:
Can we just use `MAX_BLOCK_REORG_HEIGHT` ?

Having the same constant with two names could be confusing.